src/valueObjects/errors/NotUnderEditorialWorkflowError.js

```diff
@@ -0,0 +1,16 @@
+import BaseError from './BaseError';
+
+/**
+ * Error thrown when posts are not a part of an editorial workflow. 
+ */
+export default class NotUnderEditorialWorkflowError extends BaseError {
```

Comment:
I really like the separation of `NotUnderEditorialWorkflowError` from `APIError`, and checking the type of an error seems like a much better approach than checking a property on an error. Great improvements here.